app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -19,14 +25,25 @@ open class FeedViewModel : ViewModel() {
     fun getFeedItems(): LiveData<List<FeedItem>> = feedItems
     fun getNetworkErrorEvent(): LiveData<Event<String>> = networkErrorEvent
 
-    init {
+    fun initRepo(feedRepository: FeedRepository) {
+        this.feedRepository = feedRepository
+        feedItems.addSource(feedRepository.getFeedItems(), Observer { t ->
+            isEmpty.value = t.isEmpty()
+            feedItems.value = t
+        })
         refresh()
     }
 
+    @SuppressLint("CheckResult")
     fun refresh() {
-        //todo: fix the implementation
-        isLoading.value = false
-        isEmpty.value = true
+        isLoading.value = true
+        feedRepository.refresh().subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                isLoading.value = false
+            }, { error ->
+                networkErrorEvent.value = Event(error.toString())
```

Comment:
Should we disable `isLoading` in case of an error?

Reply:
Also, note that either network error or the message could be null.

Reply:
What is the suggested behavior in case the message is null?
As long as `error` isn't null, wouldn't `error.toString()` handle the case of the message being null?

Reply:
Not sure that toString() actually prints anything useful.
I suggest: `error?.message ?: "unknown error"`

Reply:
It prints the error message wrapped up nicely. Should I still change the code?

Reply:
How did you verify? Did you run the app with Airplane mode ON?

Reply:
I had errors when I was writing the classes.
Also tried running it with Airplane mode ON now and it looks great.